eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessor.java

```diff
@@ -16,26 +16,100 @@
+    private Function<ISubmittedVariant, String> hashingFunction;
 
     public SubSnpNoHgvsToDbsnpVariantsWrapperProcessor() {
-        subSnpNoHgvsToSubmittedVariantProcessor = new SubSnpNoHgvsToSubmittedVariantProcessor();
         subSnpNoHgvsToClusteredVariantProcessor = new SubSnpNoHgvsToClusteredVariantProcessor();
+        hashingFunction = new DbsnpSubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
     }
 
     @Override
     public DbsnpVariantsWrapper process(SubSnpNoHgvs subSnpNoHgvs) throws Exception {
         DbsnpVariantsWrapper dbsnpVariantsWrapper = new DbsnpVariantsWrapper();
-        dbsnpVariantsWrapper.setSubmittedVariants(subSnpNoHgvsToSubmittedVariantProcessor.process(subSnpNoHgvs));
-        dbsnpVariantsWrapper.setClusteredVariant(subSnpNoHgvsToClusteredVariantProcessor.process(subSnpNoHgvs));
-        // TODO create operations
+        List<DbsnpSubmittedVariantEntity> variants = new ArrayList<>();
+        List<DbsnpSubmittedVariantOperationEntity> operations = new ArrayList<>();
+        DbsnpClusteredVariantEntity clusteredVariant = subSnpNoHgvsToClusteredVariantProcessor.process(subSnpNoHgvs);
+
+        List<String> alternateAlleles = subSnpNoHgvs.getAlternateAllelesInForwardStrand();
+        for (String alternateAllele : alternateAlleles) {
+            SubmittedVariant variant = subSnpNoHgvsToSubmittedVariant(subSnpNoHgvs, alternateAllele);
+            if (!variant.isAllelesMatch()) {
+                decluster(subSnpNoHgvs.getSsId(), variant, operations);
+            }
+
+            String hash = hashingFunction.apply(variant);
+            DbsnpSubmittedVariantEntity ssVariantEntity = new DbsnpSubmittedVariantEntity(subSnpNoHgvs.getSsId(), hash,
+                                                                                          variant);
+            ssVariantEntity.setCreatedDate(subSnpNoHgvs.getSsCreateTime().toLocalDateTime());
+            variants.add(ssVariantEntity);
+        }
+
+        dbsnpVariantsWrapper.setDbsnpVariantType(subSnpNoHgvs.getDbsnpVariantType());
+        dbsnpVariantsWrapper.setClusteredVariant(clusteredVariant);
+        dbsnpVariantsWrapper.setSubmittedVariants(variants);
+        dbsnpVariantsWrapper.setOperations(operations);
         return dbsnpVariantsWrapper;
     }
+
+    private SubmittedVariant subSnpNoHgvsToSubmittedVariant(SubSnpNoHgvs subSnpNoHgvs, String alternate) {
+        Region variantRegion = getVariantRegion(subSnpNoHgvs);
+        String reference = subSnpNoHgvs.getReferenceInForwardStrand();
+        SubmittedVariant variant = new SubmittedVariant(subSnpNoHgvs.getAssembly(), subSnpNoHgvs.getTaxonomyId(),
+                                                        getProjectAccession(subSnpNoHgvs),
+                                                        variantRegion.getChromosome(), variantRegion.getStart(),
+                                                        reference, alternate, subSnpNoHgvs.getRsId());
```

Comment:
Calling the constructor without flags will set their default values, even though at this point we don't know if it matches the assembly or not. I think it would be better to use the constructor will all the arguments, and setting the assembly match as null.

We could even reorder the constructor parameters to something like supported by evidence, validated, alleles match and assembly match. This will place the null at the very end and kind of replicate the application flow, but could cause issues in the tests if the refactoring doesn't reorder the input automatically.

Reply:
`assemblyMatch` can't be null in `SubmittedVariantEntity` so it will be set to `false` but the real value will need to be determined by the assembly checker in a later step